spring-rabbit/src/main/java/org/springframework/amqp/rabbit/retry/MissingMessageIdAdvice.java

```diff
@@ -40,8 +40,11 @@
  * Must be placed before the retry interceptor in the advice chain.
  * @author Gary Russell
  * @since 1.1.2
+ * @deprecated since 1.7 in favor of non-retryable messages logic in the {@code StatefulRetryOperationsInterceptor}
  *
  */
+@Deprecated
+@SuppressWarnings("deprecation")
```

Comment:
We can't just deprecate it; it simply won't work any more with spring-retry 1.2.